### PR TITLE
add pod e2e testing

### LINKED ISSUE #76: Add e2e test in Kosmos
**What would you like to be added**:
add e2e testing code for pod, deployment
**Why is this needed**:

### PR DESCRIPTION
**Pull request by kuangcheng66**
#### What type of PR is this?
test: add test codes

#### What does this PR do?
add pod e2e testing code

#### Which issue(s) does this PR fix?
Fixes #76 

#### Screenshots (if applicable):
```

```

#### Additional Comments:
```

```

### CODE REVIEW

File: test/e2e/framework/cluster.go [hunk 1]
```diff
@@ -108,3 +111,17 @@ func runCmd(command string, args ...string) error {
 
 	return nil
 }
+
+// GetKnodeClient get knode client
+func GetKnodeClient(client versioned.Interface, clusterName string) kubernetes.Interface {
```
Comment (zhouhaoA1) [inline test/e2e/framework/cluster.go]: parameter `clusterName` make me confused,  `knodeName` maybe  more suitable

File: test/e2e/framework/constant.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package framework
+
+import "time"
+
+const (
+	// pollInterval defines the interval time for a poll operation.
+	pollInterval = 5 * time.Second
+	// pollTimeout defines the time after which the poll operation times out.
+	pollTimeout = 420 * time.Second
```
Comment (zhouhaoA1) [inline test/e2e/framework/constant.go]: 420s is too long, e2e should close to real usage.

File: test/e2e/pod_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			podName = "pod-e2e-ns-" + rand.String(3)
+			pod = helper.NewPod(knodeName, namespaceName, podName)
+			framework.CreatePod(kubeClient, pod)
+		})
+
+		ginkgo.It("pod should be disappeared on knodes", func() {
+			framework.WaitPodDisappearOnKnodes(kosmosClient, knodeName, namespaceName, podName)
+		})
+	})
+})
```
Comment (zhouhaoA1) [inline test/e2e/pod_test.go:64]: nice test case.  Test create pod in kosmos and specify pod NodeName , then check the result by immediately connect the knode apiserver.

File: test/e2e/framework/cluster.go (hunk 1 above)
Comment (kuangcheng66) [reply test/e2e/framework/cluster.go]: ok，i will fix it

File: test/e2e/framework/constant.go (hunk 2 above)
Comment (kuangcheng66) [reply test/e2e/framework/constant.go:9]: ok

Comment (kuangcheng66) [reply test/e2e/framework/constant.go:9]: 420 is set with reference to Karamda.

File: test/e2e/pod_test.go (hunk 3 above)
Comment (kuangcheng66) [reply test/e2e/pod_test.go:64]: thank you

File: test/e2e/framework/constant.go (hunk 2 above)
Comment (kuangcheng66) [reply test/e2e/framework/constant.go]: ok

**Pull request closed without merging**